packages/cardano-services/test/load/load.test.ts

```diff
@@ -0,0 +1,288 @@
+    });
+    proc.once('error', reject);
+  });
+
+const stopProc = (proc: ChildProcess) =>
+  new Promise<void>((resolve) => (proc?.kill() ? proc.on('close', resolve) : resolve()));
+
+let serverProc: ChildProcess;
+let workerProc: ChildProcess;
+
+const startServer = async (options: TestOptions = {}) => {
+  await setupRabbitMQContainer(containerName);
+  if (env.START_LOCAL_HTTP_SERVER)
+    serverProc = await runCli(
+      [
+        'start-server',
+        '--api-url',
+        configEnv.TX_SUBMIT_HTTP_URL,
+        ...(options.directlyToOgmios ? [] : ['--use-queue']),
+        ...commonArgs,
+        ServiceNames.TxSubmit
+      ],
+      '"msg":"Started'
+    );
+};
+
+const startWorker = async (options: TestOptions = {}) => {
+  workerProc = await runCli(
+    [
+      'start-worker',
+      ...commonArgs,
+      ...(options.parallel ? ['--parallel', '--parallel-txs', env.WORKER_PARALLEL_TRANSACTION.toString()] : [])
+    ],
+    '"msg":"TxSubmitWorker: starting'
+  );
+};
+
+const stopServer = () => stopProc(serverProc);
+const stopWorker = () => stopProc(workerProc);
+
+describe('load', () => {
+  const testReports: TestReport[] = [];
+
+  let wallet: ObservableWallet;
+  let address: Cardano.Address;
+
+  const prepareWallets = async () => {
+    wallet = await getWallet();
+    ({ address } = (await firstValueFrom(wallet.addresses$))[0]);
+
+    const startTime = Date.now();
+    await firstValueFrom(wallet.syncStatus.isSettled$.pipe(filter((isSettled) => isSettled)));
+    logger.info(`Time to settle wallets ${Date.now() - startTime}ms`);
+  };
+
+  const waitForTxInBlockchain = (txId: Cardano.TransactionId) =>
+    firstValueFrom(wallet.transactions.history$.pipe(filter((txs) => txs.filter((tx) => tx.id === txId).length === 1)));
+
+  const fragmentWhenRequired = async (options: TestOptions) => {
+    const utxo = await firstValueFrom(wallet.utxo.available$);
+    const coins = 1_000_000n + 1000n * BigInt(env.TRANSACTIONS_NUMBER - 1);
+    const requiredUtxo = env.TRANSACTIONS_NUMBER * 3 + 1;
+    const suitableUtxo = utxo.filter(([, txOut]) => txOut.value.coins >= coins);
+    const haveEnoughUtxo = suitableUtxo.length >= requiredUtxo;
+
+    if (haveEnoughUtxo) return;
+
+    const fragment = async () => {
+      const output = { address, value: { coins } };
+      const tx = await wallet.initializeTx({
+        // eslint-disable-next-line unicorn/no-new-array
+        outputs: new Set([...new Array(requiredUtxo)].map((_) => ({ ...output })))
+      });
+
+      logger.info(`Fragmentation tx: ${tx.hash}`);
+
+      await wallet.submitTx(await wallet.finalizeTx(tx));
+      await waitForTxInBlockchain(tx.hash);
+    };
+
+    if (options.directlyToOgmios) await fragment();
+    else {
+      await Promise.all([startWorker(), fragment()]);
+      await stopWorker();
+    }
+  };
+
+  beforeAll(async () => {
+    jest.setTimeout(180_000);
+
+    // Dependencies setup parallelization
```

Comment:
I suggest to `startServer` first and `prepareWallets` only after services are healthy. We don't want wallet to deal with failed requests (and take more time as a result of exponential backoff) when using db-sync providers to fetch the data.